dump.go

```diff
@@ -21,9 +22,30 @@ func withColor(color int, text string) string {
 }
 
 func writeBody(buf *bytes.Buffer, req *http.Request) error {
-	body, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		return err
+	var body []byte
+	var err error
+
+	if strings.Contains(req.Header.Get("Accept-Encoding"), "gzip") {
+		gz, err := gzip.NewReader(req.Body)
```

Comment:
@gchaincl i will try to avoid additional body read with this behaviour, give me a lil time for it.